tests/test_dataset_generator.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+from unittest import mock
+
+import pandas as pd
+from numpy.random import PCG64, Generator
+
+from synthetic_data.dataset_generator import generate_dataset_by_class
+
+
+class TestDatasetGenerator(unittest.TestCase):
+    def setUp(self):
+        self.rng = Generator(PCG64(12345))
+        self.dataset_length = 10
+
+    def test_generate_dataset_with_invalid_generator(self):
+        columns_to_gen = [{"generator": "non existent generator"}]
+        with self.assertRaisesRegex(
+            ValueError, "generator: non existent generator is not a valid generator."
+        ):
+            generate_dataset_by_class(
+                self.rng,
+                columns_to_generate=columns_to_gen,
+                dataset_length=self.dataset_length,
+                path=None,
+            )
+
+    @mock.patch("synthetic_data.dataset_generator.logging.warning")
+    def test_generate_dataset_with_none_columns(self, mock_warning):
+        empty_dataframe = pd.DataFrame()
+        df = generate_dataset_by_class(self.rng, None, self.dataset_length, None)
+        mock_warning.assert_called_once_with(
+            "columns_to_generate is empty, empty dataframe will be returned."
+        )
+        self.assertEqual(empty_dataframe.empty, df.empty)
+
+    def test_generate_custom_datasets(self):
+        columns_to_gen = [
+            {"generator": "integer", "min_value": 0, "max_value": 1},
+            {"generator": "datetime", "start_date": pd.Timestamp(2049, 12, 31)},
+            {
+                "generator": "text",
+                "chars": ["a", "b", "c"],
+                "str_len_min": 300,
+                "str_len_max": 301,
+            },
+        ]
+        expected_columns = ["integer", "datetime", "text"]
+        df = generate_dataset_by_class(
+            self.rng,
+            columns_to_generate=columns_to_gen,
+            dataset_length=self.dataset_length,
+            path=None,
+        )
+        self.assertEqual(self.dataset_length, len(df))
+        self.assertListEqual(list(df.columns), expected_columns)
```

Comment:
since we are setting seed and passing the generator, should we assert the values of the array?

Reply:
we won't need to assert len, etc then since it is an exact array.

Reply:
I think in this case its okay to assertEqual on the length of the dataset and separately check the columns name of the expected and the actual are the same -- two separate checks that are valid to check